docs/src/appendix/snforge-library/fs.md

```diff
@@ -2,9 +2,91 @@
 
 Module containing functions for interacting with the filesystem.
 
-* [`read_txt`](fs/read_txt.md) - reads and parses plain text file content into an array of felts
-* [`parse_txt`](fs/parse_txt.md) - parses plain text file content and tries to deserialize it into the
-  specified type
-* [`read_json`](fs/read_json.md) - reads and parses json file content into an array of felts
-* [`parse_json`](fs/parse_json.md) - parses json file content and tries to deserialize it into the specified
-  type
\ No newline at end of file
+## `File`
+```
+trait FileTrait {
+    fn new(path: ByteArray) -> File;
+}
+```
+
+## `FileParser<T>`
+```rust
+trait FileParser<T, impl TSerde: Serde<T>> {
+    fn parse_txt(file: @File) -> Option<T>;
+    fn parse_json(file: @File) -> Option<T>;
+}
+```
+
+## `read_txt` & `read_json`
+```rust
+fn read_txt(file: @File) -> Array<felt252>;
+fn read_json(file: @File) -> Array<felt252>;
+```
+
+
+## File format
```

Comment:
As an end goal we probably would like to put those in cairo, yup?

Reply:
I thought of a guide here actually, but both would work